single_kernel_mongo/managers/mongos_operator.py

```diff
@@ -125,6 +135,11 @@ def on_secret_changed(self, secret_label: str, secret_id: str) -> None:
 
     @override
     def on_config_changed(self) -> None:
+        """We check the expose-external value as it can be invalid.
+
+        And then, update external services, update TLS certificates and share
+        connection information with client if it changed.
+        """
```

Comment:
```
Handle configurations for expose-external.

It is necessary to check that the option is valid, and if it has changed we must update external services, update TLS certificates and share
        connection information with client. This is because when we change our connectivity we update the IP address of mongos. 
```